src/java.base/share/classes/jdk/internal/reflect/Reflection.java

```diff
@@ -106,6 +106,13 @@ public static void ensureMemberAccess(Class<?> currentClass,
         }
     }
 
+    public static void ensureNativeAccess(Class<?> currentClass) {
+        Module module = currentClass.getModule();
+        if (!SharedSecrets.getJavaLangAccess().isEnableNativeAccess(module)) {
+            throw new IllegalCallerException("Illegal native access from module: " + module);
```

Comment:
You may want to drop "module" from the exception message as Module::toString is specified to return "module XXXX" or "unnamed module XXXX" so you'll end up duplication "module" in the message.